com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java

```diff
@@ -461,10 +505,35 @@ private void sendRetry() {
         outputFrame(ashFrame);
     }
 
+    private boolean isStatusRequest(AshFrame frame) {
+        int magicNumber = 126;
+        int[] buffer = frame.getOutputBuffer();
+
+        int[] rawDataBuffer = frame.dataBuffer;
+
+        // AshFrameAck
+        if (buffer.length == 4 && buffer[3] == magicNumber) {
```

Comment:
Are you sure this works across all versions of ASH? I need to look at this more, but why are you peeking into the raw buffer rather than the frame type? Maybe it's not decoded at that point? I think it would be much better to use the decoded frame, otherwise I don't think this code works with the different versions of ASH.

Reply:
Yes, right. Its not decoded at this point. If I do `frame.getOutputBuffer()` I get nothing for this type (AshFrameAck) of paket.

That's why I was asking for your help, since its your code :)

Regarding the layer issue: I thought that chips connected via SPI cannot be detached anyway, so there is no need for such a heartbeat. Also I looked at the types of FrameHandlers and there is the one for SPI and the one for ASH, they do not have a common Base class. Can you point me to the EZSP "frame handling" class where you imagine the  heartbeat should be placed?

But anyhow, if you cannot detach a SPI chip, do you think we need that heartbeat? Because all we could tell the user is: "Your chipset is broken, it doesn't respond"...